packages/react-graphql/src/hooks/apollo-context.tsx

```diff
@@ -0,0 +1,43 @@
+import ApolloClient from 'apollo-client';
+import React from 'react';
+
+const ApolloContext = React.createContext<null | ApolloClient<any>>(null);
+
+export interface ApolloProviderProps<CacheShape> {
+  readonly children?: React.ReactNode;
+  readonly client: ApolloClient<CacheShape>;
+}
+
+export function ApolloProvider<CacheShape = any>({
+  client,
+  children,
+}: ApolloProviderProps<CacheShape>): React.ReactElement<
+  ApolloProviderProps<CacheShape>
+> {
+  return (
+    <ApolloContext.Provider value={client}>{children}</ApolloContext.Provider>
```

Comment:
If we are already making a custom component for this, can we render the actual one from apollo here? If not, then can we just export `ApolloContext` directly? I'd like to be more consistent with this and generally the full context object is the most useful export since it lets you do stuff with `useContext` as well as giving access to both the provider and consumer.